### PR TITLE
Add slot-level targeting to Opt Out

### PR DESCRIPTION
**Pull request by chrislomaxjones**
## What does this change?

Add slot-level targeting to Opt Out advertising. We insert the targeting parameter just before we define each of the individual slots.

We match the slot targeting we send to GAM as closely as possible. Example targeting:
- For slot `dfp-ad--top-above-nav` we add `slot=top-above-nav`.
- For slot `dfp-ad--inline1` we add `slot=inline1`.
- For slot `dfp-ad--inline1-mobile` we add `slot=inline1`.

Note we now pass the slot name directly into the `defineSlot` function in order to supply it as the targeting value. We rename the value of type `'inline' | 'inline-right'` to `slotKind` to represent the _kind_ of slot it is. In a subsequent PR we'll remove this parameter in favour of computing Opt Out compatible names directly.

## Does this change need to be reproduced in dotcom-rendering ?

- [X] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## Screenshots

Excerpt of the request we make to Opt Out. Note the `customs` object associated with each slot contains the new targeting.

Screenshot 2022-11-22 at 14 38 40

## What is the value of this and can you measure success?

Campaigns can be targeted to specific slots on our page.

### Tested

- [X] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/consentless/init-fixed-slots.ts [hunk 1]
```diff
@@ -10,11 +10,10 @@ const initFixedSlots = (): Promise<void> => {
...
 	// define slots
 	adverts.forEach((slotElement) => {
-		const slotName = slotElement.dataset.name?.includes('inline')
-			? 'inline'
-			: slotElement.dataset.name;
+		const slotName = slotElement.dataset.name;
+		const slotKind = slotName?.includes('inline') ? 'inline' : undefined;
 		if (slotName) {
-			defineSlot(slotElement, slotName);
+			defineSlot(slotElement, slotName, slotKind);
```
Comment (Jakeii) [inline static/src/javascripts/projects/commercial/modules/consentless/init-fixed-slots.ts:16]: Although a bit more verbose, I think separate `defineFixedSlot `and `defineInlineSlot` or similar might be better?

**Review by Jakeii — APPROVED** [review]
Sweet!

File: static/src/javascripts/projects/commercial/modules/consentless/define-slot.ts [hunk 2]
```diff
@@ -2,12 +2,32 @@ import { log } from '@guardian/libs';
...
 
-	const filledCallback: OptOutFilledCallback = (_adSlot, response) => {
+	const filledCallback: OptOutFilledCallback = (
+		_adSlot,
+		{ width, height },
+	) => {
 		log('commercial', `Filled consentless ${slotId}`);
-		const { width, height } = response;
+
 		if (width === 1 && height === 1) {
```
Comment (arelra) [inline static/src/javascripts/projects/commercial/modules/consentless/define-slot.ts:31]: Minor

```suggestion
    const isFluid = width === 1 && height === 1;
		if (isFluid) {
```

**Review by arelra — APPROVED** [review]

File: static/src/javascripts/projects/commercial/modules/consentless/define-slot.ts (hunk 2 above)
Comment (chrislomaxjones) [reply static/src/javascripts/projects/commercial/modules/consentless/define-slot.ts]: Good shout, thanks!

File: static/src/javascripts/projects/commercial/modules/consentless/init-fixed-slots.ts (hunk 1 above)
Comment (chrislomaxjones) [reply static/src/javascripts/projects/commercial/modules/consentless/init-fixed-slots.ts:16]: I really like this idea. I'm going to hold off on this PR as I think the interface for `defineSlot` is going to change once when we change the slot names, and we can implement something along these lines then.

**Pull request merged**